### DISCUSSION TITLE
Understandable file organization

### DISCUSSION THREAD

**Discussion by Arutosio** [ppy/osu#17275] [Ideas] [answered]
could you do an understandable file organization like on osu!, with folder for beatmap and beatmap content?
for example like this:
```
beatmaps
 |- beatmap(folder)
 |  |- screen
 |  |- video
 |  `- song
 |- beatmap(folder)
 |  |- screen
 |  |- video
 |  `- song
 `- beatmap(folder)
    |- screen
    |- video
    `- song
```
in this way it would be possible to find the backgrounds of the songs, audio files and videos.

**Comment by frenzibyte** [comment, marked as answer]
~~It's intentionally made in a way that prevents users from accessing it directly, since such files should not be interfered with.~~ You may wanna have a read on https://github.com/ppy/osu/wiki/User-file-storage for more information.

**Reply by peppy** [reply]
It was not "intentionally made to prevent users from accessing". It was done for efficiency and allowing sharing of files between beatmaps, skins, etc.

If you word it like that you are only going to piss people off.

**Reply by Arutosio** [reply]
i hope someone will make sure to implement a way to export the mp3, video or background image file just by right clicking from the beatmap @peppy 
![image](https://user-images.githubusercontent.com/30377429/158574987-3d77a707-654c-48fa-9541-7a6ac76df64a.png)

**Reply by bdach** [reply]
@arutosio sorry but i don't think this will be supported. i don't know why you would want these options but i can't see any feasible good use case for them.

**Reply by Sublimelime** [reply]
There are some use cases, though you might disagree on whether or not they're good:

- Pulling the background because you like the image, to be used (eg) as a desktop background
- Pulling the mp3 to be listened to outside of osu (though this kinda makes osu a music distributor of sorts, which makes the copyright lawyer in me sweat a bit)
- Pulling the video to be watched outside of osu (same potential issue as above though)

It does somewhat seem unnecessary, as you can just export the entire beatmap then unzip it to get at the components. Adding a direct piecemeal export would just save the unzipping step.

**Reply by bdach** [reply]
use cases 2 and 3 are not going to be supported. osu! is a rhythm game and not a source of music/video files. 1 is also not far from these other ones in that respect.

and maps can already be exported via editor.

**Comment by peppy** [comment]
@Arutosio accessing the files of a beatmap can be done from the editor already (this should be the main time you need to access them). You can export them from the file menu. In the future the editor will also allow you to create a folder with the current beatmap's files for better accessibility, ie. if you want to add or modify files using an external application.
